packages/foam-vscode/src/extension.ts

```diff
@@ -120,6 +125,73 @@ async function updateReferenceList() {
   }
 }
 
+async function openDailyNote() {
+  const foamConfiguration = workspace.getConfiguration("foam");
+  const currentDate = new Date();
+
+  const dailyNotePath = getDailyNotePath(foamConfiguration, currentDate);
+
+  createDailyNoteIfNotExists(foamConfiguration, dailyNotePath, currentDate);
+  await focusDailyNote(dailyNotePath);
+}
+
+function getDailyNotePath(configuration: WorkspaceConfiguration, date: Date) {
+  const rootDirectory = workspace.workspaceFolders[0].uri.fsPath;
+  const dailyNoteDirectory: string =
+    configuration.get("openDailyNote.directory") ?? ".";
+  const dailyNoteFilename = getDailyNoteFileName(configuration, date);
+
+  return join(rootDirectory, dailyNoteDirectory, dailyNoteFilename);
+}
+
+function getDailyNoteFileName(
+  configuration: WorkspaceConfiguration,
+  date: Date
+): string {
+  const filenameFormat: string = configuration.get(
+    "openDailyNote.filenameFormat"
+  );
+  const fileExtension: string = configuration.get(
+    "openDailyNote.fileExtension"
+  );
+
+  return `${dateFormat(date, filenameFormat, false)}.${fileExtension}`;
+}
+
+function createDailyNoteIfNotExists(
+  configuration: WorkspaceConfiguration,
+  dailyNotePath: string,
+  currentDate: Date
+) {
+  if (existsSync(dailyNotePath)) {
```

Comment:
Is there a reason to use `*Sync` APIs? The parent functions are all async, so we could just await on the async `fs.promises.*` methods?